File: first-unique-character-in-a-string/first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        HashMap<Character, Integer> map = new HashMap<Character, Integer>();
        char[] sChars = s.toCharArray();
        for(Character c : sChars){
            int count = map.getOrDefault(c, 0);
            map.put(c, count + 1);
        }
        for(Character c : sChars){
            if(map.get(c) == 1) return s.indexOf(c);
        }
        return -1;
    }
}

File: first-unique-character-in-a-string/first-unique-character-in-a-string.kt
class Solution {
    fun firstUniqChar(s: String): Int {
        val map = HashMap<Char, Int>()
        
        val sArray = s.toCharArray()
        for(c in sArray){
            map[c] = map.getOrDefault(c, 0) + 1
        }
        
        for(i in 0 until sArray.size){
            if( map[ sArray[i] ] == 1 ) return i
        }
        
        return -1
    }
}